File: 프로그래머스/2/68645. 삼각 달팽이/삼각 달팽이.py
def solution(n):
    answer = []
    board = [[0] * n for i in range(n)]
    x, y, cnt = -1, 0, 0
    num = 1
    
    for i in range(n, 0, -1):
        if cnt % 3 == 0:
            for j in range(i):
                x += 1
                board[x][y] = num
                num += 1
                
        elif cnt % 3 == 1:
            for j in range(i):
                y += 1
                board[x][y] = num
                num += 1
        
        else:
            for j in range(i):
                x -= 1
                y -= 1
                board[x][y] = num
                num += 1
                
        cnt += 1
        
    for i in range(n):
        for j in range(i + 1):
            answer.append(board[i][j])
                
    return answer

File: 프로그래머스/2/68645. 삼각 달팽이/삼각 달팽이.java
class Solution {
    public int[] solution(int n) {
        int[][] snail = new int[n][n];
        
        int x = -1;
        int y = 0;
        int num = 1;
        
        for(int i=0; i<n; i++) {
            for(int j=i; j<n; j++) {
                if(i % 3 == 0) {
                    x++;
                } else if(i % 3 == 1) {
                    y++;
                } else {
                    x--;
                    y--;
                }
                snail[x][y] = num++;
            }
        }
        
        int idx = 0;
        int[] answer = new int[num-1];
        
        for(int i=0; i<n; i++) {
            for(int j=0; j<n; j++) {
                if(snail[i][j] == 0)
                    break;
                answer[idx++] = snail[i][j];
            }
        }
        
        return answer;
    }
}